tendermint/src/serializers.rs

```diff
@@ -185,3 +188,34 @@ where
         }))
     }
 }
+
+pub(crate) fn parse_non_empty_id<'de, D>(deserializer: D) -> Result<Option<Id>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    if s.is_empty() {
+        Ok(None)
+    } else {
+        // TODO: how can we avoid rewriting code here?
+        match Id::from_str(&s).map_err(|_| {
+            D::Error::custom(format!(
+                "expected {}-character hex string, got {:?}",
+                LENGTH * 2,
+                s
+            ))
+        }) {
+            Ok(id) => Ok(Option::from(id)),
+            Err(_) => Ok(None),
```

Comment:
In case there is an error above, we just return `Ok(None)`. That doesn't seem right unless we want to ignore it. But then the return could be changed from `Result<Option<Id>, D::Error>` to `Option<Id>`.

Reply:
I'm going to try to find a way to get away from this custom serialization completely. The whole problem of `Option` is here because we again assume that the value of CommitSig.validator_address can be `null`.

Reply:
This is related then: https://github.com/informalsystems/tendermint-rs/pull/196/files#r399823439

Reply:
I've been trying to figure out better ways of handling serialization - especially with the input from Ethan et al. I think we should wrap up this PR as soon as possible without adding even more features and break out any finds like this into their own issue. Some of the solutions I found would require the refactoring of the serializers library, which will lead to a cleaner code, but it would add more complexity on top of this PR.

Also, getting away from custom serialization is a tall order, but we can put some of the complexity into the serializers so the user of the structs can use them without any precursor.